subsys/net/l2/ethernet/lldp/lldp.c

```diff
@@ -249,6 +249,55 @@ static int lldp_start(struct net_if *iface, u32_t mgmt_event)
 	return 0;
 }
 
+int net_lldp_recv(struct net_if *iface, struct net_pkt *pkt)
+{
+	struct ethernet_context *ctx;
+	net_lldp_recv_callback_func_t cb;
+	int ret;
+
+	ret = lldp_check_iface(iface);
+	if (ret < 0) {
+		return NET_DROP;
+	}
+
+	ctx = net_if_l2_data(iface);
+
+	ret = lldp_find(ctx, iface);
+	if (ret < 0) {
+		return NET_DROP;
+	}
+
+	cb = ctx->lldp[ret].cb;
+	if (cb) {
+		cb(iface, pkt);
```

Comment:
I just wonder if we should do "return cb(iface, pkt)" here. This way the callback is responsible to free the packet and it won't accidentally try to access the pkt that would be freed otherwise.
So the callback would need to return:
* NET_DROP, if packet was invalid or rejected etc. In this case the core stack will free the packet.
* NET_OK, if the packet was accepted, in this case the ownership of the net_pkt goes to callback and core network stack will forget it.

So the above should be added to callback documentation, and callbacks return value should be "enum net_verdict". Also the sample app would need to be changed to free the packet after properly parsing it.

Reply:
OK, I will return net_verdict.